prow/report/report.go

```diff
@@ -107,22 +111,35 @@ func Report(ghc GithubClient, reportTemplate *template.Template, pj v1.ProwJob)
 	if ghc == nil {
 		return fmt.Errorf("trying to report pj %s, but found empty github client", pj.ObjectMeta.Name)
 	}
+
 	if !pj.Spec.Report {
 		return nil
 	}
+
 	refs := pj.Spec.Refs
-	if len(refs.Pulls) != 1 {
-		return fmt.Errorf("prowjob %s has %d pulls, not 1", pj.ObjectMeta.Name, len(refs.Pulls))
+	if len(refs.Pulls) > 1 {
+		return fmt.Errorf("prowjob %s has %d pulls, thus will skip report", pj.ObjectMeta.Name, len(refs.Pulls))
 	}
+
 	childDescription := fmt.Sprintf("Waiting on: %s", pj.Spec.Context)
 	if err := reportStatus(ghc, pj, childDescription); err != nil {
 		return fmt.Errorf("error setting status: %v", err)
 	}
+
 	// Report manually aborted Jenkins jobs and jobs with invalid pod specs alongside
 	// test successes/failures.
 	if !pj.Complete() {
 		return nil
 	}
+
+	if len(refs.Pulls) == 0 {
```

Comment:
I'd maybe separate out these two scenarios:

```golang
if len(r.P) == 0 && pj.S.T == v1.PJ {
  return fmt.Ef()
}
if len(r.P) == 0 {
  return nil
}
```

Reply:
My brain melted trying to read this